components/Navbar/Navbar.js

```diff
@@ -27,8 +28,28 @@ export default function Navbar() {
     };
   }, []);
 
+  useEffect(() => {
+    const handleScroll = () => {
+      if (window.pageYOffset > 0) {
+        setIsScrolled(true);
+      } else {
+        setIsScrolled(false);
+      }
+    };
+    window.addEventListener("scroll", handleScroll);
+    return () => {
+      window.removeEventListener("scroll", handleScroll);
+    };
+  }, []);
+
+  const boxShadowStyle = isScrolled
+    ? {
+        boxShadow: `0 1px 2px 0 rgb(var(--foreground-rgb)), 0 1px 3px 1px rgb(var(--foreground-rgb))`,
+      }
+    : {};
+
   return (
-    <div className={styles.root}>
+    <div className={styles.root} style={boxShadowStyle}>
```

Comment:
you can concat className like this `className={${styles.root} ${isScrolled && styles.scrolled}}`

in module.css
```
.root.scrolled {
  box-shadow: 0 1px 2px 0 rgb(var(--foreground-rgb)), 0 1px 3px 1px rgb(var(--foreground-rgb));
  /* ... */
}
```